lib/Onyx.ts

```diff
@@ -594,6 +594,42 @@ function clear(keysToPreserve: OnyxKey[] = []): Promise<void> {
         .then(() => undefined);
 }
 
+function updateSnapshots(data: OnyxUpdate[]) {
+    const snapshotCollectionKey = OnyxUtils.getSnapshotKey();
+    if (!snapshotCollectionKey) return;
+
+    const promises: Array<() => Promise<void>> = [];
+    const finalPromise = Promise.resolve();
+
+    const snapshotCollection = OnyxUtils.getCachedCollection(snapshotCollectionKey);
+
+    data.forEach(({key, value}) => {
+        // snapshots are normal keys so we want to skip update if they are written to Onyx
+        if (OnyxUtils.isCollectionMemberKey(snapshotCollectionKey, key)) {
+            return;
+        }
+
+        Object.entries(snapshotCollection).forEach(([snapshotKey, snapshotValue]) => {
+            // Snapshots may not be present in cache. We don't know how to update them so we skip.
+            if (!snapshotValue) {
+                return;
+            }
+
+            if (typeof snapshotValue !== 'object' || !('data' in snapshotValue)) return;
+
+            const snapshotData = snapshotValue.data;
+            if (!snapshotData || !snapshotData[key]) {
+                return;
+            }
+
+            const updated = _.pick(value, Object.keys(snapshotData[key]));
```

Comment:
We would like to avoid using lodash/underscore. Would it be possible to code this line in plain TS?

Reply:
@blazejkustra it's possible but it will probably looks something like:
```ts
const updated = Object.keys(snapshotData[key]).reduce<Record<string, unknown>>(
  (acc, k) => {
    if (!value || typeof value !== "object") {
      return acc;
    }

    if (k in value) {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-expect-error
      acc[k] = value[k];
    }

    return acc;
  },
  {}
);
```

Reply:
Let's use lodash then:
```ts
import lodashPick from 'lodash/pick';
```